.github/workflows/tests.yml

```diff
@@ -51,14 +51,18 @@ jobs:
         working-directory: api
 
     steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v4
+
+      - name: Fetch branches
+        run: git fetch --all
+
       - name: Set up Go
-        uses: actions/setup-go@v2
-        with:
-          go-version: ^1.22
+        uses: actions/setup-go@v5
         id: go
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
+        with:
+          go-version-file: ${{ github.workspace }}/api/go.mod
+          cache: false
 
       - name: Cache Go dependencies
         uses: actions/cache@v2
```

Comment:
I see you updated this to `v4` in the comment. Do you want to do the same here?

Reply:
As it's already updating the workflow, I think we could update all dependencies to the latest version.

Reply:
I thought that it is better to create another PR for that to not mix things together: in addition to updating the versions I planned to migrate at least independent serverless tools from my repo here as well to increase the code testing and lining level